### PR TITLE
feat(stdlib): Add `==` operator to `Queue` module

### PR DESCRIPTION
**Pull request by spotandjake**
This pr adds `Queue.equal` which provides a more useful equality check that cares about the `queue` contents rather than internal structure.

### CODE REVIEW

**Comment by ospencer** [comment]
I agree with @alex-snezhko that we should name it `(==)`.

**Comment by spotandjake** [comment]
> I agree with @alex-snezhko that we should name it `(==)`.

Made that change

File: stdlib/queue.gr [hunk 1]
```diff
@@ -164,6 +164,33 @@ provide let copy = queue => {
   { size, array: Array.copy(array), headIndex, tailIndex }
 }
 
+/**
+ * Checks if two queues are equivalent by value.
+ *
+ * @param queue1: The first queue to compare
+ * @param queue2: The second queue to compare
+ *
```
Comment (phated) [inline stdlib/queue.gr:172]:
```suggestion
```

**Comment by spotandjake** [comment]
Applied that suggestion

**Comment by phated** [comment]
@spotandjake please resolve the conflicts 🙏

**Review by phated — APPROVED** [review]

**Comment by phated** [comment]
@spotandjake it looks like this conflicts again. Can you update?

**Comment by spotandjake** [comment]
Updated

**Pull request merged**